components/support/rust-logs/src/foreign_logger.rs

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Logger interface for foreign code
+//!
+//! This is what the application code defines.  It's responsible for taking rust log records and
+//! feeding them to the application logging system.
+
+pub use log::Level;
+
+/// log::Record, except it exposes it's data as fields rather than methods
+#[derive(Debug, PartialEq, Eq)]
+pub struct Record {
+    pub level: Level,
+    pub target: String,
+    pub message: String,
+}
+
+pub trait Logger: Sync + Send {
```

Comment:
`Log` sounds good to me, I'll rename.  I think we would want to wrap the `log` crate as much as possible, but wrapping the `Record` type doesn't seem right to me since it only exposes it's data through methods.  It seems better to send the data over as owned data in a struct.